### PR TITLE
Overhaul instance list header [WD-1892]

### PR DESCRIPTION
**Pull request by lorumic**

### CODE REVIEW

File: static/js/util/instanceOptions.tsx [hunk 1]
```diff
@@ -1,10 +1,10 @@
 export const instanceTypeOptions = [
   {
-    label: "Container",
+    label: "Containers",
```
Comment (edlerd) [inline static/js/util/instanceOptions.tsx:3]: This label is used for the selector on instance creation. There we need the singular form. So maybe pluralize it in place or have two of those?

File: static/js/pages/instances/InstanceList.tsx [hunk 2]
```diff
@@ -218,127 +217,133 @@ const InstanceList: FC = () => {
...
+        <div className="p-panel__header">
+          <Row className="p-panel__title p-instance-header">
+            <Col size={2}>
+              <h4>
+                {visibleInstances.length}{" "}
+                {visibleInstances.length === 1 ? "Instance" : "Instances"}
+              </h4>
+            </Col>
+            <Col size={8}>
+              <Row>
```
Comment (edlerd) [inline static/js/pages/instances/InstanceList.tsx:231]: I think we don't need the nesting here and can remove the two layers for 
```
            <Col size={8}>
              <Row>
```

File: static/js/pages/instances/InstanceList.tsx [hunk 3]
```diff
@@ -218,127 +217,133 @@ const InstanceList: FC = () => {
...
+                    options={[
+                      {
+                        label: "Containers and VMs",
+                        value: "any",
+                      },
+                      ...instanceTypeOptions,
+                    ]}
+                    value={type}
+                  />
+                </Col>
```
Comment (edlerd) [inline static/js/pages/instances/InstanceList.tsx:274]: The filters and the button look a little funny with screen width between 1000 and 1200 px.

I think we should give the button a col size of 3 at least. The filter and search can be together in one col with size 7. Then the two filters can have a fixed width and the search bar can adjust its width to fill the col-7. Maybe consult with design, as the responsive behaviour is not specified. But from the screens, I assume the filters are fixed width and the search input adjusts.

Comment (lorumic) [reply static/js/pages/instances/InstanceList.tsx]: Ok, let's first hear from @al-bakalova if your assumptions are correct, and then, once the responsive behaviour becomes clearer, I'll adapt the sizes accordingly.

**Review by edlerd — APPROVED** [review]
LGTM

**Comment by al-bakalova** [comment]
Looks good to me as well. Thank you!

[The following issue ](https://warthogs.atlassian.net/browse/WD-1915)has been created to further improve the "create new" button's behaviour.

**Pull request merged**